### DISCUSSION TITLE
Recipe Sections - Annotate different components of a dish in one recipe

### DISCUSSION THREAD

**Discussion by camillerimichael** [mealie-recipes/mealie#3348] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

Complex recipes tend to categorise different components (sometimes with repeated ingredients) in one recipe. 

Typical examples might be a protein with a glaze or sauce, garnish for soups or a cake for dessert with a section for the buttercream, another for frosting and a ganache. 

### Please provide a concise description of the feature that would resolve your issue.

The recipe schema could be augmented to include an array of sections, with each section having a title, ingredients and method, possibly even the prep and active time as well as a make ahead flag. This could be useful for marinades and ganaches to let cool down.

### Please consider and list out some caveats or tradeoffs made in your design decision

To be frank, I have not explored mealie enough yet to provide useful info here. I'm hoping to familiarise myself more and in the meantime, hoping this serves as a good point of discussion.

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by michael-genson** [comment]
You can accomplish this currently with ingredient sections and recipe sections.

For ingredients: hit the three dots to the right of the ingredient and click "Toggle Section":
![image](https://github.com/mealie-recipes/mealie/assets/71845777/3d75ff47-d3c1-4a13-b735-fb0b373acec2)

For instructions, similarly, there's a "Toggle Section" button:
![image](https://github.com/mealie-recipes/mealie/assets/71845777/25fb5cb6-725a-4a75-998c-657f8171f901)

Once you add sections they look like this:
![image](https://github.com/mealie-recipes/mealie/assets/71845777/236a0485-9d2b-4f8d-997f-17f6e985f9ff)
![image](https://github.com/mealie-recipes/mealie/assets/71845777/b5e20025-86f3-452a-8399-1efd623e551c)
